app/experimenter/base/tests/test_views.py

```diff
@@ -0,0 +1,20 @@
+from django.conf import settings
+from django.test import TestCase
+from django.urls import reverse
+
+
+class TestContextProcessors(TestCase):
+
+    def test_render_google_analytics_or_not(self):
+        headers = {settings.OPENIDC_EMAIL_HEADER: "user@example.com"}
```

Comment:
Even though it is the default, I'd prefer this block be in a `with self.settings(...)` block so that the settings this test rely on are explicit.

Reply:
That's an interesting topic. There is a `settings_test.py` which hardcodes a bunch of "defaults" specifically for the tests. 

So, when you're hacking on a test function like this you either have to memorize (or look up frequently) the contents of `settings_test.py`. *Or*, to save the code reader we *could* do things like set-what's-already-set here in the test to make it ultra explicit. 

I doubt there's a perfect solution. In lots of Django projects there are a bunch of assumptions that basically requires that you study the fixtures and/or the `settings.Test` class. 

E.g. https://github.com/mozilla-services/tecken/blob/9ea366093ee56f5e2607d1fe73ef90bb5d1ba8d2/tests/test_upload.py#L139-L150
This test could instead generate a zip on the fly there in the test instead of relying on knowing details.